resourcegroup.tf

```
@@ -0,0 +1,6 @@
+resource "azurerm_resource_group" "rg" {
+  location = var.location
+  name     = format("%s-%s-%s", var.project, var.env, "rg")
```

should be project not product, and no need to use format, use string interpolation

```suggestion
  name     = "${var.product}-${var.env}-rg"
```